### PR TITLE
RCAL-905: Add ePSF, ABVegaOffset, and ApCorr Datamodels

### PR DESCRIPTION
**Pull request by PaulHuwe**
Resolves [RCAL-905](https://jira.stsci.edu/browse/RCAL-905)

Closes #905

This PR adds datamodels and tests for ePSF, ABVegaOffset, and ApCorr reference files. Other tests were tweaked for this support. The PR template was also updated to match the one from the RAD library.

## Tasks
- [x] update or add relevant tests
- [ ] update relevant docstrings and / or `docs/` page
- [x] Does this PR change any API used downstream? (if not, label with `no-changelog-entry-needed`)
  - [x] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types)
  - [x] [start a `romancal` regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/romancal.yml) with this branch installed (`"git+https://github.com/<fork>/roman_datamodels@<branch>"`)

<details><summary>news fragment change types...</summary>

- ``changes/<PR#>.feature.rst``: new feature
- ``changes/<PR#>.bugfix.rst``: fixes an issue
- ``changes/<PR#>.doc.rst``: documentation change
- ``changes/<PR#>.removal.rst``: deprecation or removal of public API
- ``changes/<PR#>.misc.rst``: infrastructure or miscellaneous change
</details

### CODE REVIEW

**Comment by PaulHuwe** [comment]
RCAL regression tests: https://github.com/spacetelescope/RegressionTests/actions/runs/11097495928

File: src/roman_datamodels/maker_utils/_ref_files.py [hunk 1]
```diff
@@ -128,6 +234,37 @@ def mk_distortion(*, filepath=None, **kwargs):
...
+    """
+    if len(shape) != 4:
+        shape = (3, 9, 361, 361)
+        warnings.warn("Input shape must be 4D. Defaulting to (3, 9, 361, 361)")
+
+    epsfref = stnode.EpsfRef()
+    epsfref["meta"] = mk_ref_epsf_meta(**kwargs.get("meta", {}))
+
+    epsfref["psf"] = kwargs.get("psf", np.zeros(shape, dtype=np.float32))
+    epsfref["extended_psf"] = kwargs.get("extended_psf", np.zeros(shape[2:], dtype=np.float32))
```
Comment (schlafly) [inline src/roman_datamodels/maker_utils/_ref_files.py:263]: I left most of the relevant comments in the rad PR; this all looks good.  But one dumb comment here: in the actual files, is the extended_psf shape actually the same as in the psf stamps, or significantly larger?  I hope larger, even if we don't do that for the dummy epsf objects?

Comment (PaulHuwe) [reply src/roman_datamodels/maker_utils/_ref_files.py:263]: The one sample file I have does not include the `extended_psf` (it is an optional array keyword). By default, the maker utility just uses a dummy shape and is not representative of reality.

File: pyproject.toml [hunk 2]
```diff
@@ -18,8 +18,9 @@ dependencies = [
     "gwcs >=0.19.0",
     "numpy >=1.22",
     "astropy >=5.3.0",
-    "rad >= 0.21.0",
+    # "rad >= 0.21.0",
     # "rad @ git+https://github.com/spacetelescope/rad.git",
+    "rad @ git+https://github.com/PaulHuwe/rad.git@RAD-175_PSFetalSchema",
```
Comment (PaulHuwe) [inline pyproject.toml:23]:
```suggestion
    "rad @ git+https://github.com/spacetelescope/rad.git",
```

**Pull request merged**